### ISSUE TITLE
Cannot vendor crates into Mozilla because of workspace inheritance

### ISSUE THREAD

**Issue by eqrion** [bytecodealliance/wasm-tools#816]
Mozilla vendors in some of the wasm-tools crates into our source tree (using `cargo vendor` plus some glue). I recently tried to do this again for the most recent versions, and ran into an issue stemming from #783.

Essentially the wast/wasm-smith/wasm-encoder crates are copied into a directory `third_party/rust/`, but the top-level workspace Cargo.toml from this repo is not brought over. This causes errors when trying to resolve dependencies.

I believe there's an upstream `cargo vendor` issue [here](https://github.com/rust-lang/cargo/issues/11192), but no resolution yet.

I have a hack to workaround this by manually copying a trimmed version of the wasm-tools workspace Cargo.toml into our `third_party/rust` folder. But I'm not sure if it's landable as-is, because it will need to be manually updated with every new vendor and would conflict with other crates if they start using workspace inheritance.

@alexcrichton do you have any ideas of anything we can change in wasm-tools to help here? Otherwise we may need to temporarily fork until `cargo vendor` is updated, or try to fix cargo vendor ourselves.

**Comment by eqrion** [comment]
Actually, the upstream issue specifically mentions git dependencies, which I only used because a release hadn't happened yet. I'll double check with the latest release.

**Comment by eqrion** [comment]
Yup, sure enough vendoring from crates.io (instead of git) to versions published after #783 works just fine. I'll close this then.

**Comment by alexcrichton** [comment]
Good to hear crates.io works, but let us know if you need git and I'm sure something can be arranged.
